comp/otelcol/otlp/configcheck/configcheck.go

```diff
@@ -52,12 +52,27 @@ func ReadConfigSection(cfg config.Reader, section string) *confmap.Conf {
 		if strings.HasPrefix(key, prefix) && cfg.IsSet(key) {
 			mapKey := strings.ReplaceAll(key[len(prefix):], ".", confmap.KeyDelimiter)
 			// deep copy since `cfg.Get` returns a reference
-			stringMap[mapKey] = deepcopy.Copy(cfg.Get(key))
+			var val interface{}
+			if _, ok := intConfigs[key]; ok {
+				val = deepcopy.Copy(cfg.GetInt(key)) // ensure to get an int even if it is set as a string in env vars
+			} else {
+				val = deepcopy.Copy(cfg.Get(key))
+			}
+			stringMap[mapKey] = val
 		}
 	}
 	return confmap.NewFromStringMap(stringMap)
 }
 
+// intConfigs has the known config keys that may need a type cast to int by calling cfg.GetInt
+var intConfigs = map[string]struct{}{
```

Comment:
We might want to double check with Agent Configuration, but I believe this type cast to int can be done with `config.BindEnvAndSetDefault` if we set the defaults to ints. For example, I added the following change in pkg/config/setup/otlp.go and didn't see the int conversion error:
```
config.BindEnvAndSetDefault("otlp_config.receiver.protocols.grpc.max_recv_msg_size_mib", 10)
```

Reply:
This seems to have some strange side effects. Once I do this, the other otel gRPC & HTTP configs break. Will take a closer look tmr

Reply:
I'll add @hush-hush to the list of reviewers as he might have suggestions.

Reply:
@liustanley is correct. When using `Get` the config will try, as best as possible, to cast to the default type. The implementation from [viper](https://github.com/DataDog/viper/blob/master/viper.go#L651) only support basic types (int, float, time, slice of string, ..).

Reply:
> Adding a default to all those setting

What if we don't have defaults for some of those configs? https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/config.md It seems OTel requires some of them to not have a default

Reply:
@hush-hush unfortunately we cannot do this because OTLP ingestion relies on a check on whether the `otlp_config.receiver` section is present to determine whether to enable the OTLP ingestion pipeline http://github.com/DataDog/datadog-agent/blob/5c3c17908079421d4b663c6dea60b2965efe623f/comp/otelcol/otlp/configcheck/configcheck.go#L72

As soon as default values are added to any config under `otlp_config.receiver`, it will force the OTLP ingestion pipeline to be always enabled, that will be a big breaking change

Reply:
I agree this could be taken care in another PR. Can we create a card to track this effort ? We're starting to accumulate a lot of workaround around the OTLP initalization logic.